File: 1590593189-1676[1]/remoting/Marshalling/By Reference/logServer.vb
Imports System
Imports System.Runtime.Remoting
Imports System.Runtime.Remoting.Channels
Imports System.Runtime.Remoting.Channels.Http
Imports System.Runtime.Remoting.Channels.Tcp
Imports Wrox4923.Remoting.Log

Namespace Wrox4923.Remoting

    Public Class logServer
    
        Public Shared Sub Main() 
        
            ChannelServices.RegisterChannel(New HttpChannel(8000))
            ChannelServices.RegisterChannel(New TcpChannel(8001))
            
            RemotingConfiguration.RegisterWellKnownServiceType(GetType(Logger), _
            	"Log", WellKnownObjectMode.Singleton)

            Console.WriteLine("Log server listening...")
            Console.WriteLine("Press enter to stop the server...")
            Console.ReadLine()
        
        End Sub
    
    End Class

End Namespace

File: 1590593189-1675[1]/remoting/Marshalling/By Reference/logServer.cs
using System;
using System.Runtime.Remoting;
using System.Runtime.Remoting.Channels;
using System.Runtime.Remoting.Channels.Http;
using System.Runtime.Remoting.Channels.Tcp;
using Wrox4923.Remoting.Log;

namespace Wrox4923.Remoting
{
    public class logServer
    {
        public static void Main() 
        {
            ChannelServices.RegisterChannel(new HttpChannel(8000));
            ChannelServices.RegisterChannel(new TcpChannel(8001));
            
            RemotingConfiguration.RegisterWellKnownServiceType(typeof(Logger),
            	"Log", WellKnownObjectMode.Singleton);

            Console.WriteLine("Log server listening...");
            Console.WriteLine("Press enter to stop the server...");
            Console.ReadLine();
		}
	}
}